packages/amplify-graphql-transformer-core/src/utils/model-datasource-strategy-utils.ts

```diff
@@ -141,6 +141,27 @@ export const isSqlModel = (ctx: DataSourceStrategiesProvider, typename: string):
   return isSqlDbType(modelDataSourceType.dbType);
 };
 
+/**
+ * Checks if the given model is a PostgreSQL model
+ * @param ctx Transformer Context
+ * @param typename Model name
+ * @returns boolean
+ */
+export const isPostgresModel = (ctx: DataSourceStrategiesProvider, typename: string): boolean => {
+  if (isBuiltInGraphqlType(typename)) {
+    return false;
+  }
+  const modelDataSourceType = getModelDataSourceStrategy(ctx, typename);
+  return isPostgresDbType(modelDataSourceType.dbType);
+};
+
+/**
+ * Type predicate that returns true if `dbType` is a PostgreSQL database type
+ */
+export const isPostgresDbType = (dbType: ModelDataSourceStrategyDbType): dbType is ModelDataSourceStrategySqlDbType => {
```

Comment:
The return value of this function is misleading. You already know that `dbType` is a `ModelDataSourceStrategyDbType` because it's specified in the argument type. What the predicate actually does is assert that dbType is exactly equal to `POSTGRES`. Change the type predicate to assert the constant string:
```suggestion
export const isPostgresDbType = (dbType: ModelDataSourceStrategyDbType): dbType is 'POSTGRES' => {
```